service/resource/cloudformation/delete.go

```diff
@@ -3,14 +3,76 @@ package cloudformation
 import (
 	"context"
 
+	"github.com/aws/aws-sdk-go/aws"
+	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
+
+	"github.com/giantswarm/aws-operator/service/key"
 )
 
 func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	deleteStackInput, err := toDeleteStackInput(deleteChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	stackName := deleteStackInput.StackName
+	if *stackName != "" {
+		_, err := r.awsClient.DeleteStack(&deleteStackInput)
+		if err != nil {
+			return microerror.Maskf(err, "deleting AWS CloudFormation Stack")
+		}
+
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "deleting AWS CloudFormation stack: deleted")
+	} else {
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "deleting AWS CloudFormation stack: already deleted")
+	}
```

Comment:
I <3 the logging. The `r.logger.Log("cluster", key.ClusterID(customObject), "debug", "deleting AWS CloudFormation stack")` on the top is missing though.